server/src/main/java/org/elasticsearch/index/MergePolicyConfig.java

```diff
@@ -111,17 +111,33 @@ public final class MergePolicyConfig {
     private final Logger logger;
     private final boolean mergesEnabled;
     private volatile Type mergePolicyType;
+    private final ByteSizeValue defaultMaxMergedSegment;
+    private final ByteSizeValue defaultMaxTimeBasedMergedSegment;
 
     public static final double DEFAULT_EXPUNGE_DELETES_ALLOWED = 10d;
     public static final ByteSizeValue DEFAULT_FLOOR_SEGMENT = new ByteSizeValue(2, ByteSizeUnit.MB);
     public static final int DEFAULT_MAX_MERGE_AT_ONCE = 10;
     public static final ByteSizeValue DEFAULT_MAX_MERGED_SEGMENT = new ByteSizeValue(5, ByteSizeUnit.GB);
+    public static final Setting<ByteSizeValue> DEFAULT_MAX_MERGED_SEGMENT_SETTING = Setting.byteSizeSetting(
+        "merge.policy.default.max_merged_segment",
```

Comment:
I'd prefer something like:

```suggestion
        "indices.merge.policy.max_merged_segment",
```

though adding `default` could also be fine by me: `indices.default.merge.policy.max_merged_segment`.

Reply:
I pushed 63a93ecc0a8a3ce0aac91ac327d75da940105748 to use `indices.merge.policy.max_merged_segment`